### PR TITLE
Advise reporter producer

### PR DESCRIPTION
**Pull request by pacospace**
## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Kafka producer for advise reporter messages

### CODE REVIEW

File: advise_reporter.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+_LOGGER = logging.getLogger("thoth.advise_reporter")
+
+_KAFKA_BOOTSTRAP_SERVERS = os.getenv("KAFKA_BOOTSTRAP_SERVERS", "localhost:9092")
+_KAFKA_CAFILE = os.getenv("KAFKA_CAFILE", "ca.crt")
+KAFKA_CLIENT_ID = os.getenv("KAFKA_CLIENT_ID", "thoth-messaging")
+KAFKA_PROTOCOL = os.getenv("KAFKA_PROTOCOL", "SSL")
+KAFKA_TOPIC_RETENTION_TIME_SECONDS = 60 * 60 * 24 * 45
+ssl_context = ssl.create_default_context(purpose=ssl.Purpose.SERVER_AUTH, cafile=_KAFKA_CAFILE)
+app = faust.App("thoth-messaging", broker=_KAFKA_BOOTSTRAP_SERVERS, ssl_context=ssl_context, web_enabled=False)
```
Comment (KPostOffice) [inline advise_reporter.py]: import message base and use `app` which is defined there.

See https://github.com/thoth-station/package-update-job/pull/26/files#diff-698189911dcaf56ccf71e87af0104edcR34-R43

then you can delete lines  32-38

File: messages/advise_justification.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
```
Comment (KPostOffice) [inline messages/advise_justification.py]: messaging stuff is fine for now, but eventually we should use `thoth-messaging` by changing its structure to what we have here and in `package-update`

**Review by KPostOffice — COMMENTED** [review]
Looks good so far, just a few minor things

File: messages/advise_justification.py (hunk 2 above)
Comment (pacospace) [reply messages/advise_justification.py]: is there a card for that?

Comment (KPostOffice) [reply messages/advise_justification.py]: Hmm, I don't think so

Comment (fridex) [reply messages/advise_justification.py]: > messaging stuff is fine for now, but eventually we should use `thoth-messaging` by changing its structure to what we have here and in `package-update`

+1

File: advise_reporter.py (hunk 1 above)
Comment (fridex) [inline advise_reporter.py]: This looks like shared configuration for all the components that will use thoth-messaging. Can this be logic placed into one place - `thoth-messaging` module itself?

Comment (fridex) [reply advise_reporter.py]: The same way as the logging is setup for all the components that log. The setup should be lazy not to increase time of imports.

File: messages/advise_justification.py (hunk 2 above)
Comment (pacospace) [reply messages/advise_justification.py]: https://trello.com/c/6wBPlbUU let's do it!!

File: advise_reporter.py (hunk 1 above)
Comment (KPostOffice) [reply advise_reporter.py]: https://github.com/thoth-station/messaging/pull/100/files#diff-f262f1b1016b774b837759e34a9781d0R29-R38

Comment (pacospace) [reply advise_reporter.py]: Yes thanks :) , as soon as that PR is merged and new thoth-messaging is released, I will use that functionality and remove the redundant part here.

**Comment by pacospace** [comment]
Issue installing `thoth-lab` and `thoth-messaging` for too tight constraint on dependency jinja2:
```
  pandas-profiling==2.5.0   requires astropy>=3.2.3, confuse==1.0.0, htmlmin==0.1.12, ipywidgets==7.5.1, jinja2==2.11.1, kaggle==1.5.6, matplotlib>=3.0.3, missingno==0.4.2, numpy>=1.16.0, pandas==0.25.3, phik==0.9.9, requests==2.22.0, scipy>=1.4.1, tangled-up-in-unicode==0.0.3, tqdm==4.42.0, visions==0.2.2
 
  thoth-messaging==0.3.2    requires aiocontextvars==0.2.2, aiohttp-cors==0.7.0, aiohttp==3.6.2, argo-workflows==2.1.4, async-timeout==3.0.1, attrdict==2.0.1, attrs==19.3.0, cachetools==4.0.0, certifi==2019.11.28, chardet==3.0.4, click==7.0, colorclass==2.2.0, colorlog==4.1.0, contextvars==2.4, croniter==0.3.31, daiquiri==2.0.0, dictdiffer==0.8.1, faust==1.10.2, google-auth==1.11.0, idna-ssl==1.1.0, idna==2.8, immutables==0.11, jinja2==3.0.0a1, kafka-python==1.4.7, kubernetes==10.0.1, markupsafe==1.1.1, mock==4.0.1, mode==4.3.1, multidict==4.7.4, mypy-extensions==0.4.3, oauthlib==3.1.0, openshift==0.10.1, opentracing==1.3.0, pyasn1-modules==0.2.8, pyasn1==0.4.8, python-dateutil==2.8.1, python-json-logger==0.1.11, python-string-utils==0.6.0, pytz==2019.3, pyyaml==5.3, requests-oauthlib==1.3.0, requests==2.22.0, rfc5424-logging-handler==1.4.3, robinhood-aiokafka==1.1.4, rsa==4.0, ruamel.yaml.clib==0.2.0, ruamel.yaml==0.16.10, sentry-sdk==0.14.1, six==1.14.0, terminaltables==3.1.0, thoth-common==0.10.6, typing-extensions==3.7.4.1, tzlocal==2.1b1, urllib3==1.25.8, venusian==1.2.0, websocket-client==0.57.0, yarl==1.4.2
```
`pipenv.patched.piptools.exceptions.NoCandidateFound: Could not find a version that matches jinja2==2.11.1,==3.0.0a1`

One solution is to avoid pre releases in thoth-messaging
```
[pipenv]
allow_prereleases = false
```

@fridex WDYT?

**Comment by fridex** [comment]
> ```
> [pipenv]
> allow_prereleases = false
> ```
> 
> @fridex WDYT?

+1, we should not use pre-releases as these apps sooner or later end up in stage+prod.

**Review by fridex — APPROVED** [review]
LGTM

**Pull request merged**